src/types/dmabuf.rs

```diff
@@ -1,23 +1,39 @@
 //! Support for the DMABuf type
 
-use wayland_sys::server::wl_display as wl_server_display;
-use wlroots_sys::{wl_display, wlr_renderer, wlr_linux_dmabuf_v1, 
-    wlr_linux_dmabuf_v1_create, wlr_linux_dmabuf_v1_destroy};
+use wlroots_sys::{wl_display, wlr_linux_dmabuf_v1, wlr_linux_dmabuf_v1_create, 
+                wlr_linux_dmabuf_v1_destroy};
+use {compositor::Compositor,
+    render::GenericRenderer};
+
 
 #[derive(Debug)]
 pub struct Dmabuf {
     dmabuf: *mut wlr_linux_dmabuf_v1
 }
 
 impl Dmabuf {
-    pub(crate) unsafe fn new(display: *mut wl_server_display, renderer: *mut wlr_renderer) -> Option<Self> {
-        let dmabuf_raw = wlr_linux_dmabuf_v1_create(display as *mut wl_display, renderer);
-        if !dmabuf_raw.is_null() {
-            Some(Dmabuf { dmabuf: dmabuf_raw })
-        } else {
-            None
+    pub fn new(compositor: &Compositor) -> Option<Self> {
+        unsafe {
+            // Get the renderer from compositor
+            let renderer: Option<GenericRenderer> = compositor.renderer;
+            // Unwrap it and check if it's null
+            let renderer_raw = renderer.unwrap().as_ptr();
+            if !renderer_raw.is_null() {
```

Comment:
This null check is unnecessary, but it's not obvious because GenericRenderer's internal pointer isn't in a `NonNull`. I opened #294 to address that. For now you can remove the is null check.